main.tf

```diff
@@ -345,3 +345,62 @@ module "deployment_user" {
 
   source = "./modules/deployment_user"
 }
+
+# ---------------------------------------------------------------------------------------------------------------------
+# scheduled HTTPS API call
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "eventbridge" {
+  count = var.enable_scheduled_api_call ? 1 : 0
+
+  source     = "terraform-aws-modules/eventbridge/aws"
+  create_bus = false
+
+  rules = {
+    crons = {
+      description         = "Trigger for a Lambda"
+      schedule_expression = var.scheduled_api_call_crontab
+    }
+  }
+
+  targets = {
+    crons = [
+      {
+        name  = "lambda-https-cron"
+        arn   = module.lambda[0].lambda_function_arn
+        input = jsonencode({ "url" : var.scheduled_api_call_url, "httpVerb" : "GET" })
+      }
+    ]
+  }
+}
+
+data "archive_file" "function" {
+  count       = var.enable_scheduled_api_call ? 1 : 0
+  output_path = "${path.module}/index.zip"
+  source_file = "${path.module}/index.js"
+  type        = "zip"
+}
+
+# tfsec:ignore:aws-lambda-enable-tracing
+module "lambda" {
+  count = var.enable_scheduled_api_call ? 1 : 0
+
+  source  = "terraform-aws-modules/lambda/aws"
+  version = "~> 2.0"
+
+  function_name = "terra3-https-api-call"
+  description   = "scheduled https api call"
+  handler       = "index.handler"
+  runtime       = "nodejs16.x"
+
+  create_package         = false
+  local_existing_package = data.archive_file.function[0].output_path
```

Comment:
Don't create the zip by yourself with data.archive_file but rather use what's already given by the lambda module, namely the "source_path". Move up the index.js as well into the scheduled_api_call folder and rename it to something more self-explaining, e.g. api_call.js